docker/docker.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package docker exposes an API to convert Jujuisms to dockerisms.
+package docker
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/juju/deputy"
+	"gopkg.in/juju/charm.v5"
+)
+
+var execCommand = exec.Command
+
+// Launch runs a new docker container with the given process data.
+func Launch(p charm.Process) (ProcDetails, error) {
+	args, err := launchArgs(p)
+	if err != nil {
+		return ProcDetails{}, err
+	}
+	d := deputy.Deputy{
+		Errors: deputy.FromStderr,
+	}
+	cmd := execCommand("docker", args...)
+	out := &bytes.Buffer{}
+	cmd.Stdout = out
+	if err := d.Run(cmd); err != nil {
+		return ProcDetails{}, err
+	}
+	id := string(bytes.TrimSpace(out.Bytes()))
+	status, err := inspect(id)
+	if err != nil {
+		return ProcDetails{}, fmt.Errorf("can't get status for container %q: %s", id, err)
+	}
+	return ProcDetails{ID: status.Name, ProcStatus: ProcStatus{Status: status.brief()}}, nil
+}
+
+// Status returns the ProcStatus for the docker container with the given id.
+func Status(id string) (ProcStatus, error) {
+	status, err := inspect(id)
+	if err != nil {
+		return ProcStatus{}, err
+	}
+	return ProcStatus{Status: status.brief()}, nil
+}
+
+// Destroy stops and removes the docker container with the given id.
+func Destroy(id string) error {
```

Comment:
Presumably there is more clean-up that should go on here.  From the way folks have been talking, there are all sorts of things that have to manually cleaned up that the plugin should take care of instead.